frontend/src/routes/Search.tsx

```diff
@@ -15,41 +22,84 @@ import { SmallDescription } from "../ui/metadata";
 import { Card } from "../ui/Card";
 import { Breadcrumbs, BreadcrumbsContainer, BreadcrumbSeparator } from "../ui/Breadcrumbs";
 import { MissingRealmName } from "./util";
-import { ellipsisOverflowCss } from "../ui";
+import { ellipsisOverflowCss, focusStyle } from "../ui";
 import { COLORS } from "../color";
 import { BREAKPOINT_MEDIUM, BREAKPOINT_SMALL } from "../GlobalStyle";
-import { keyOfId } from "../util";
+import { isExperimentalFlagSet, keyOfId } from "../util";
+import { Button } from "../ui/Button";
 import { DirectVideoRoute, VideoRoute } from "./Video";
 
 
 export const isSearchActive = (): boolean => document.location.pathname === "/~search";
 
+export type Item = Exclude<ItemType, "%future added value">;
+type SearchParams = {
+    query: string;
+    itemType?: Item;
+    start?: string;
+    end?: string;
+}
+
 export const SearchRoute = makeRoute({
-    url: ({ query }: { query: string }) => `/~search?${new URLSearchParams({ q: query })}`,
+    url: ({ query, itemType, start, end }: SearchParams) => {
+        const searchParams = new URLSearchParams({ q: query });
+
+        if (itemType) {
+            searchParams.append("f", itemType);
+        }
+        if (start) {
+            searchParams.append("start", start);
+        }
+        if (end) {
+            searchParams.append("end", end);
+        }
+
+        return `/~search?${searchParams}`;
+    },
     match: url => {
         if (url.pathname !== "/~search") {
             return null;
         }
 
         const q = url.searchParams.get("q") ?? "";
-        const queryRef = loadQuery<SearchQuery>(query, { q });
+        const filters = prepareFilters(url);
+
+        const queryRef = loadQuery<SearchQuery>(query, { q, filters });
 
         return {
             render: () => <RootLoader
                 {...{ query, queryRef }}
                 noindex
                 nav={() => []}
-                render={data => <SearchPage q={q} outcome={data.search} />}
+                render={data => <SearchPage {...{ q }} outcome={data.search} />}
             />,
             dispose: () => queryRef.dispose(),
         };
     },
 });
 
+const itemTypes: Item[] = ["EVENT", "REALM"];
+export const validType = (value: string | null) => itemTypes.includes(value as Item)
+    ? value as Item
+    : null;
```

Comment:
Technically the `value as Item` cast there is incorrect. However, in practice it really doesn't matter. Still, this would be more clean: `(itemTypes as string[]).includes(value)`.